presto-native-execution/presto_cpp/main/operators/tests/BroadcastTest.cpp

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/Uri.h>
+#include "folly/init/Init.h"
+#include "presto_cpp/external/json/json.hpp"
+#include "presto_cpp/main/operators/BroadcastWrite.h"
+#include "presto_cpp/main/operators/LocalPersistentShuffle.h"
+#include "presto_cpp/main/operators/tests/PlanBuilder.h"
+#include "velox/buffer/Buffer.h"
+#include "velox/exec/Exchange.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/exec/tests/utils/QueryAssertions.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/serializers/PrestoSerializer.h"
+
+using namespace facebook::velox;
+using namespace facebook::presto;
+using namespace facebook::presto::operators;
+
+namespace facebook::presto::operators::test {
+class BroadcastTest : public exec::test::OperatorTestBase {
+ public:
+ protected:
+  void SetUp() override {
+    exec::test::OperatorTestBase::SetUp();
+    filesystems::registerLocalFileSystem();
+    exec::Operator::registerOperator(
+        std::make_unique<BroadcastWriteTranslator>());
+  }
+
+  static std::string makeTaskId(const std::string& prefix, int num) {
+    auto url = fmt::format("batch/{}-{}", prefix, num);
+    return url;
+  }
+
+  std::shared_ptr<exec::Task> makeTask(
+      const std::string& taskId,
+      core::PlanNodePtr planNode,
+      int destination) {
+    auto queryCtx = std::make_shared<core::QueryCtx>(executor_.get());
+    core::PlanFragment planFragment{planNode};
+    return exec::Task::create(
+        taskId, std::move(planFragment), destination, std::move(queryCtx));
+  }
+
+  void runBroadcastTest(const std::vector<RowVectorPtr>& data) {
+    exec::Operator::registerOperator(
+        std::make_unique<BroadcastWriteTranslator>());
+
+    auto dataType = asRowType(data[0]->type());
+    auto writerTaskId = makeTaskId("leaf", 0);
+    auto basePath = exec::test::TempDirectoryPath::create()->path;
+    auto writerPlan = exec::test::PlanBuilder()
+                          .values(data, true)
+                          .addNode(addBroadcastWriteNode(basePath))
+                          .planNode();
+
+    exec::test::CursorParameters params;
+    params.planNode = writerPlan;
+    auto [taskCursor, results] = readCursor(params, [](auto /*task*/) {});
+
+    // Expect one file for each request.
+    ASSERT_EQ(results.size(), 1);
+
+    // Validate file path prefix is consistent.
+    auto broadcastFilePath =
+        results.back()->childAt(0)->as<SimpleVector<StringView>>()->valueAt(0);
+    ASSERT_EQ(broadcastFilePath.str().find(basePath), 0);
+
+    // Read back broadcast data from broadcast file.
+
+    // Assert data from broadcast file matches input.
```

Comment:
Perhaps, move // Assert comment after reading data and just before assertEqualResults.